File: console/src/main/kotlin/app/accrescent/parcelo/console/data/net/Edit.kt
// Copyright 2023 Logan Magee
//
// SPDX-License-Identifier: AGPL-3.0-only

package app.accrescent.parcelo.console.data.net

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Edit(
    val id: String,
    @SerialName("app_id")
    val appId: String,
    @SerialName("short_description")
    val shortDescription: String?,
    @SerialName("creation_time")
    val creationTime: Long,
    val status: EditStatus,
)

@Serializable
enum class EditStatus {
    @SerialName("unsubmitted")
    UNSUBMITTED,

    @SerialName("submitted")
    SUBMITTED,
}

File: console/frontend/src/app/edit.ts
// Copyright 2023 Logan Magee
//
// SPDX-License-Identifier: AGPL-3.0-only

export interface Edit {
    id: string;
    app_id: string;
    short_description?: string;
    creation_time: number;
    status: EditStatus;
}

export enum EditStatus {
    Unsubmitted = 'unsubmitted',
    Submitted = 'submitted',
    Rejected = 'rejected',
    Publishing = 'publishing',
    Published = 'published',
}
